ServiceNow Service Catalog for Terraform - documentation updates: new workspace paragraphs, fixing execution mode. (#520)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/service-now/service-catalog-terraform/service-catalog-config.mdx

## Execution Mode

If using v2.2.0 or above, the Service Catalog app allows you to set an [execution mode](/terraform/cloud-docs/workspaces/settings#execution-mode) for your Terraform workspaces. There are two modes to choose from:

 - The default value is "Remote", which instructs Terraform Cloud to perform runs on its disposable virtual machines.
 - Selecting "Agent" mode allows you to run Terraform operations on isolated, private, or on-premises infrastructure. This option requires you to create an Agent Pool in your organization beforehand, then provide that Agent Pool's id when you order a new workspace through the Service Catalog.

<!-- BEGIN: TFC:only name:pnp-callout -->
@include 'tfc-package-callouts/agents.mdx'
<!-- END: TFC:only name:pnp-callout -->

## Workspace Name

> Version 2.4.0 of the Service Catalog for Terraform introduces the ability to set custom names for your Terraform workspaces. Previously, workspace names defaulted to the ServiceNow RITM numbers without an option to change them. Now, you can choose a custom name that the RITM number will automatically append to, such as: `prod-us-east-1_RITM0010231`. If you leave the name blank, it will continue to default to the RITM number.

Reviewer: ```suggestion
Version 2.4.0 of the Service Catalog for Terraform introduces the ability to set custom names for your Terraform workspaces. You can choose a prefix for your workspace name that the Service Catalog app will append the ServiceNow RITM number to. If you do not define a workspace prefix, ServiceNow will use RITM number as the workspace name.
```

> Workspace names can include letters, numbers, dashes (`-`), and underscores (`_`), and should not exceed 90 characters.

Reviewer: Small nit, when we say "should not exceed 90 characters", can they technically be longer? Or is 90 the maximum length?

Author: Yes, workspace names can be very long (I tried)! They just don't look good in the UI. The 90 char recommendation comes from the TFC workspace documentation page.

> Refer to our [workspace naming recommendations](/terraform/cloud-docs/workspaces/creating#workspace-naming) for best practices.

Reviewer: ```suggestion
Refer to the [workspace naming recommendations](/terraform/cloud-docs/workspaces/creating#workspace-naming) for best practices.
```

## Workspace Tags

> Version 2.4.0 also allows you to add custom tags to Terraform workspaces. Use the new "Workspace Tags" field to provide a comma-separated list of tags that will be parsed and attached to the workspace in Terraform Cloud. 

Reviewer: ```suggestion
Version 2.4.0 also allows you to add custom tags to your Terraform workspaces. Use the "Workspace Tags" field to provide a comma-separated list of tags that will be parsed and attached to the workspace in Terraform Cloud. 
```

Tags give admins an easier way to categorize, filter, and manage workspaces provisioned through the Service Catalog for Terraform.

> We recommend that admins work with their end users to set naming conventions for tags to avoid variations like `ec2`, `aws-ec2`, `aws_ec2`.

Reviewer: ```suggestion
We recommend that you set naming conventions for tags with your end users to avoid variations such as `ec2`, `aws-ec2`, `aws_ec2`.
```
